cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,48 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
 
+import SignUpPageObject from '../support/pages/signUp.pageObject';
+
+const signUpPage = new SignUpPageObject();
+
 describe('Sign Up page', () => {
-  before(() => {
+  let user;
+
+  beforeEach(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
+
+    signUpPage.visit();
+  });
+
+  it('should provide an ability to sign up', () => {
+    signUpPage.typeUsername(user.username);
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword(user.password);
+    signUpPage.clickSignUpBtn();
+    signUpPage.checkSuccess('Welcome!');
+  });
 
+  it('should not provide an ability to sign up without username', () => {
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword(user.password);
+    signUpPage.clickSignUpBtn();
+    signUpPage.checkFail('Registration failed!');
```

Comment:
Verify that the `checkFail` method in the `SignUpPageObject` correctly checks for the error message 'Registration failed!' when the username is missing. This will ensure the test accurately detects registration failures.